### PR TITLE
Features rework

### PR DESCRIPTION
**Pull request by rustyrussell**
This is all in preparation for a `--wumbo` option.

First we update the spec, which means we gain the ability to have multiple addresses of the same type.

Then we make our feature infrastructure more flexible.  We keep each feature bitmap, and make sure every subdaemon which needs it has one.

This already helps the plugin-adds-features case, which wasn't complete before (we wouldn't accept features we advertized, since we were checking against our compiled-in feature set).

### CODE REVIEW

**Comment by cdecker** [comment]
Seems CI is just getting upset at some whitespace, should be easy to fix.

File: common/features.c [hunk 1]
```diff
@@ -126,6 +126,28 @@ void features_cleanup(void)
 	our_features = tal_free(our_features);
 }
 
+bool features_additional(const struct feature_set *newfset)
+{
+	/* Check first, before we change anything! */
+	for (size_t i = 0; i < ARRAY_SIZE(newfset->bits); i++) {
+		/* FIXME: We could allow a plugin to upgrade an optional feature
+		 * to a compulsory one? */
```
[the remark this answers is not in the document]
Comment (rustyrussell) [reply common/features.c]: Oh, are they?  I didn't see that in the code or documentation?

[the remark this answers is not in the document]
Comment (cdecker) [reply common/features.c]: My bad, I misremembered. I had considered making them odd-only initially in the hope of reducing the fallout from a plugin crashing, but must've decided against it.

**Comment by cdecker** [comment]
ACK https://github.com/ElementsProject/lightning/pull/3609/commits/e3b5ea482757fde24b1102deca1ccd1594d654f4

**Pull request merged**